scipy/interpolate/_pade.py

```diff
@@ -29,8 +32,7 @@ def pade(an, m, n=None):
     >>> e_exp = [1.0, 1.0, 1.0/2.0, 1.0/6.0, 1.0/24.0, 1.0/120.0]
     >>> p, q = pade(e_exp, 2)
 
-    >>> e_exp.reverse()
-    >>> e_poly = np.poly1d(e_exp)
+    >>> e_poly = np.polynomial.Polynomial(e_exp)
```

Comment:
That's a backcompat break, as Pamphille already said. Two possibilities: either revert back to `poly1d` (sigh) or add a separate function to return a Polynomial with your other improvements, and deprecate the older one. 
I'd recommend the latter maybe?

Reply:
I would also favor doing the latter. This further allows us to break with the convention `q.coef[0]==1` and use the QR decomposition to get the null-vector instead of solving the system. This is conceptionally cleaner and more stable, albeit slower.
In rare cases, `q.coef[0]==1` fails. (Had to exclude these in the [property tests](https://github.com/DerWeh/gftools/blob/007082a615b90d285010f4cd8e5e1ffc2ca7235e/gftool/tests/hermpade_test.py#L49) of my Pade implementation.)